odl/util/utility.py

```diff
@@ -8,25 +8,31 @@
 
 """Utilities mainly for internal use."""
 
-from __future__ import print_function, division, absolute_import
+from __future__ import absolute_import, division, print_function
+
+import inspect
+import sys
 from builtins import object
 from collections import OrderedDict
 from functools import wraps
+from future.moves.itertools import zip_longest
 from itertools import product
-import inspect
-import numpy as np
-import sys
 
+import numpy as np
 
-__all__ = ('array_str', 'dtype_str', 'dtype_repr', 'npy_printoptions',
-           'signature_string', 'indent',
-           'is_numeric_dtype', 'is_int_dtype', 'is_floating_dtype',
-           'is_real_dtype', 'is_real_floating_dtype',
-           'is_complex_floating_dtype', 'real_dtype', 'complex_dtype',
-           'is_string', 'nd_iterator', 'conj_exponent', 'writable_array',
-           'run_from_ipython', 'NumpyRandomSeed', 'cache_arguments', 'unique')
+__all__ = (
+    'array_str', 'dtype_str', 'dtype_repr', 'npy_printoptions',
+    'signature_string', 'signature_string_parts', 'repr_string',
+    'indent', 'dedent', 'attribute_repr_string', 'method_repr_string',
+    'is_numeric_dtype', 'is_int_dtype', 'is_floating_dtype', 'is_real_dtype',
+    'is_real_floating_dtype', 'is_complex_floating_dtype',
+    'real_dtype', 'complex_dtype', 'is_string', 'nd_iterator', 'conj_exponent',
+    'writable_array', 'run_from_ipython', 'NumpyRandomSeed',
+    'cache_arguments', 'unique',
+    'REPR_PRECISION')
```

Comment:
This file is getting out of hand. Perhaps we should split it (but leave that for another PR).

Reply:
See #1386